flask/namespaces.py

```diff
@@ -4,6 +4,46 @@
 from app import swagger
 from flasgger.utils import swag_from
 
+from datetime import datetime
+from auth import generate_uuid
+
+@app.route("/namespaces", methods=["POST"])
+def create_namespace():
+    uuid = request.form.get("uuid")
+
+    if not uuid:
+        return jsonify({"code": 401, "message": "Unauthorized"}), 401
+    
+    namespace_name = request.form.get("namespace")
+
+    if not namespace_name:
+        return jsonify({"code": 400, "message": "Please enter namespace name"}), 400
+    
+    # Get the user document from the uuid.
+    user_doc = db.users.find_one({"uuid": uuid})
+
+    # Get the namespace document from the namespace name.
+    # To check if already a namespace exists by this name.
+    namespace_doc = db.namespaces.find_one({"namespace": namespace_name})
+
+    # Check if namespace already exists.
+    if namespace_doc:
+        return jsonify({"code": 400, "message": "Namespace already exists"}), 400
+
+    # Generate an access token for accessing the namespace.
+    access_token = generate_uuid()
+
+    namespace_obj = {
+        "namespace": namespace_name,
+        "createdAt": datetime.utcnow(),
+        "createdBy": user_doc["_id"],
+        "accessToken": access_token
```

Comment:
I'd personally not create the `access_token` together with the namespace but decouple it (create it on demand). I'd have a list of `access_tokens` that consists of a `token` and a `createdAt` date. This way we can choose let it expire after a certain period. Also it could have a `deactivated` flag to deactivate it right away if desired. But we can also do that later.

But there is another thing. How do connect the token to the user? This token here is namespace-specific but not user-specific. Don't forget that the user is not logged in fpm. So if the someone uses this api, we don't know who it is and have to deduce it from the token. Maybe this is better:

```json
{
  "namespace": namespace_name,
  ...
  "upload_tokens": [
    {
      "token": <uuid>,
      "createdAt": <datetime>,
      "user": <uuid>,
      "deactivated": <boolean>
    }
  ]
}
```

Reply:
Thanks for the feedback @minhqdao . 

We will connect the token to the uuid (that is of the user) and from that we will be able to get the details of the user who is accessing the upload API.

So, if we don't have to create `access_token` during the creation of namespace. Can I just create a new API to generate the `access token` separately ?

And that will also be usable from the registry frontend.

Reply:
I would say so, yes. Just a need another "Create upload token" button and a `base_url/namespaces/<namespace/uploadToken` endpoint. Actually I think it's an upload token, not an access token because it is used only for the upload.

Reply:
There's probably more logic if you think it through that we could implement later. You wouldn't want to create a new upload token if there is still a valid one. If you want to create a new one, you first have to deactivate the old one.

Reply:
Cool ! Sounds perfect I think.

@henilp105 I think now you might have to create a button as well on the UI inside the namespace card for generating upload token. I will take care of the API for that.